phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -1584,12 +1588,27 @@ public static MutationMetricQueue.MutationMetric updateMutationBatchFailureMetri
                 numDeleteMutationsInBatch++;
             }
         }
+
+        long totalFailedMutation = numUpsertMutationsInBatch + numDeleteMutationsInBatch;
+        //TODO: this case is something that should not happen
+        //but still if condition makes sese if this ever happens
+        if (totalFailedMutation < numFailedMutations) {
+            LOGGER.warn(
+                    "total failed mutation less than num of failed mutation.  This is not expected.");
+            totalFailedMutation = numFailedMutations;
+        }
+        System.out.println(GLOBAL_MUTATION_BATCH_FAILED_COUNT.getMetric().getValue());
+
+        GLOBAL_MUTATION_BATCH_FAILED_COUNT.update(
+                allDeletesMutations && !isTransactional ? numDeleteMutationsInBatch :
+                        totalFailedMutation);
```

Comment:
we can extract the logic and keep value in one variable so the same variable can be used for both `GLOBAL_MUTATION_BATCH_FAILED_COUNT.update` and `MutationMetricQueue.MutationMetric`